SmartCharger.Business/Services/AuthService.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+using SmartCharger.Data.Entities;
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace SmartCharger.Business.Services
+{
+    public class AuthService
+    {
+        public string GenerateJWT(User user)
+        {
+            var securityKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes("eKCzyZdSQrr3liD89SmtxAiC7aBWiacL"));
+            var creds = new SigningCredentials(securityKey, SecurityAlgorithms.HmacSha256);
+
+            var claims = new[]
+            {
+                new Claim("userId", user.Id.ToString()),
+                new Claim("roleId", user.RoleId.ToString()),
+                new Claim(JwtRegisteredClaimNames.Name, user.FirstName)
```

Comment:
Could you explain to me what exactly is returned in "JwtRegisteredClaimNames.Name"?

Reply:
JwtRegisteredClaimNames.Name represents "name" attribute in JWT. JwtRegisteredClaimNames contains a lot of other constants that are used in claims such as: Birthdate, Email, Gender and other. This could have been replaced with just string "name".
Example of generated JWT:
```jwt
{
  "userId": "1",
  "roleId": "1",
  "name": "Leon",
  "exp": 1701535086
}
```

Reply:
Thank you for explaining.